pymongo/change_stream.py

```diff
@@ -16,48 +16,125 @@
+            raise TypeError("pipeline must be a list")
+
+        common.validate_string_or_none('full_document', full_document)
+        validate_collation_or_none(collation)
+        common.validate_non_negative_integer_or_none("batchSize", batch_size)
+
+        self._target = target
         self._pipeline = copy.deepcopy(pipeline)
         self._full_document = full_document
         self._resume_token = copy.deepcopy(resume_after)
         self._max_await_time_ms = max_await_time_ms
         self._batch_size = batch_size
         self._collation = collation
+        self._start_at_operation_time = start_at_operation_time
         self._session = session
         self._cursor = self._create_cursor()
 
-    def _full_pipeline(self):
+    @property
+    def _aggregation_target(self):
+        """The argument to pass to the aggregate command."""
+        raise NotImplementedError
+
+    @property
+    def _database(self):
+        """The database against which the aggregation commands for
+        this ChangeStream will be run. """
+        raise NotImplementedError
+
+    def _full_pipeline(self, inject_options=None):
         """Return the full aggregation pipeline for this ChangeStream."""
         options = {}
+
         if self._full_document is not None:
             options['fullDocument'] = self._full_document
         if self._resume_token is not None:
             options['resumeAfter'] = self._resume_token
+        if self._start_at_operation_time is not None:
+            options['startAtOperationTime'] = self._start_at_operation_time
+
+        if inject_options is not None:
+            options.update(inject_options)
         full_pipeline = [{'$changeStream': options}]
         full_pipeline.extend(self._pipeline)
         return full_pipeline
 
+    def _run_aggregation_cmd(self, session, explicit_session):
+        """Run the full aggregation pipeline for this ChangeStream and return
+        the corresponding CommandCursor.
+        """
+        with self._database.client._socket_for_reads(self._target._read_preference_for(session)) as (sock_info, slave_ok):
+            pipeline = self._full_pipeline()
+            cmd = SON([("aggregate", self._aggregation_target),
+                       ("pipeline", pipeline),
+                       ("cursor", {})])
+
+            result = sock_info.command(
+                self._database.name,
+                cmd,
+                slave_ok,
+                self._target._read_preference_for(session),
+                self._target.codec_options,
+                parse_write_concern_error=True,
+                read_concern=self._target.read_concern,
+                collation=self._collation,
+                session=session,
+                client=self._database.client)
+
+            cursor = result["cursor"]
+
+            if (self._start_at_operation_time is None and
+                self._resume_token is None and
+                cursor.get("_id", None) is None and
```

Comment:
get() returns None if the key doesn't exist. No need to pass a second argument.

Reply:
I was channeling the Zen of Python 😄 

["Explicit is better than implicit"](https://www.python.org/dev/peps/pep-0020/)